frontend/src/apiMocks/handlers.ts

```diff
@@ -0,0 +1,237 @@
+import { rest, RestHandler, RestRequest } from "msw";
+import { CustomAliasData, RandomAliasData } from "../hooks/api/aliases";
+import { ProfileData } from "../hooks/api/profile";
+import { RuntimeData } from "../hooks/api/runtimeData";
+import { mockIds, domainaddresses, profiles, relayaddresses } from "./mockData";
+
+export function getHandlers(
+  defaultMockId: null | typeof mockIds[number] = null
+): RestHandler[] {
+  const handlers: RestHandler[] = [];
+
+  const getMockId = (req: RestRequest): typeof mockIds[number] | null => {
+    const authHeader = req.headers.get("Authorization");
+    if (typeof authHeader !== "string") {
+      return defaultMockId;
+    }
+    const token = authHeader.split(" ")[1];
```

Comment:
4. We then look at the `Authorization` header, which normally (when running in `dev` mode) includes a user's API token to authenticate them, for a `mockId` which specifies which mock data we want to load.